os/arch/arm/src/armv7-a/arm_cpuhotplug.c

```diff
@@ -83,24 +86,41 @@ int arm_hotplug_handler(int irq, void *context, void *arg)
 			   (ie. TizenRT Kernel logic is not ready to handle such case yet)
 			   Thus, we will be facing a deadlock when cpuA fire SGI2 to cpuB (ie. pausing CPU)
 	*/
+	int cpu = this_cpu();
+	char curr_task[CORE_IDLE_TASK_CHAR_COUNT];
+	snprintf(curr_task, CORE_IDLE_TASK_CHAR_COUNT, "CPU%d IDLE", cpu);
+	/* If current cpu is in idle thread, then proceed for hotplug operation */
+	if (!strncmp(g_running_tasks[cpu]->name, curr_task, CORE_IDLE_TASK_CHAR_COUNT)) {
```

Comment:
we can directly check with task pid as idle task pid = cpu

Reply:
Thank you for suggesting, this seems more clear-cut to me.